src/batch/src/executor2/generic_exchange.rs

```diff
@@ -62,26 +65,31 @@ pub struct DefaultCreateSource {}
 impl CreateSource for DefaultCreateSource {
     async fn create_source(
```

Comment:
I originally wanted to design the create_source interface to take 
Vec<ProstExchangeSource> return Result<Vec<Box<dyn ExchangeSource>>> but found that MergeSortExchangeExecutor2 
also implements this trait, so I didn't change the trait

Reply:
Do you need me to raise another PR and refactor the poll source of MergeSortExchangeExecutor2 as well? @lmatz

Reply:
We may keep the `create_source` interface unchanged for the moment.

>Do you need me to raise another PR and refactor the poll source of MergeSortExchangeExecutor2 as well?
That would be fantastic if you are interested in it!

Reply:
I review the logic of MergeSortExchangeExecutor2. It seems to take every first data_chunk from sources and sort them, so maybe it is not necessary to refactor that?

Reply:
Yes, you are right. Let's leave it for now.